be/src/exprs/vectorized/arithmetic_expr.cpp

```diff
@@ -18,11 +21,68 @@ namespace starrocks::vectorized {
                                                       \
     virtual Expr* clone(ObjectPool* pool) const override { return pool->add(new CLASS_NAME(*this)); }
 
+static std::optional<PrimitiveType> eliminate_trivial_cast_for_decimal_mul(const Expr* e) {
+    if (!e->is_cast_expr()) {
+        return {};
+    }
+    const auto* e_child = e->get_child(0);
+    const auto& e_type = e->type();
+    const auto& e_child_type = e_child->type();
+    if (e_type.is_decimalv3_type() && e_child_type.is_decimalv3_type() && e_type.scale == e_child_type.scale) {
+        return {e_child->type().type};
+    } else {
+        return {};
+    }
+}
+
 template <PrimitiveType Type, typename OP>
 class VectorizedArithmeticExpr final : public Expr {
 public:
     DEFINE_CLASS_CONSTRUCTOR(VectorizedArithmeticExpr);
+
+    std::optional<ColumnPtr> evaluate_decimal_fast_mul(ExprContext* context, vectorized::Chunk* chunk) {
+        auto lhs_pt_opt = eliminate_trivial_cast_for_decimal_mul(_children[0]);
```

Comment:
I think we should do this logic in FE

Reply:
frankly speak, I think it is a not a good idea. arithmetic operation rules are stable to users, how to compute is intrinsic things.  Only modifying code of BE have at lest two benefits.
1. Users are not puzzled by sometimes i64 x i64=>i128,  somtimes i128 x i128 => i128. the DBA would be disturbed by users to tell why.
2. There are so many arithmetic optization cases, for examples, i8 x i8 => float32, i8 x i8 => i16, i16 x i16=>i32, in the future,if we optimize these cases, we just opt code of be.